pkg/store/prometheus.go

```diff
@@ -298,6 +298,32 @@ func labelsMatches(lset labels.Labels, ms []storepb.LabelMatcher) (bool, []store
 	return true, newMatcher, nil
 }
 
+func labelsMatchesAndTranslate(lset labels.Labels, ms []storepb.LabelMatcher) (bool, []labels.Matcher, error) {
```

Comment:
This might work after few nits, but let's focus on *reusing* stuff. We do matching almost on every component. We for sure can figure out some helper/abstraction for this somehow critical path. (: More code reused = potentially fewer bugs and less code to read. What do you think?

Reply:
Totally agree with you. I have delete the method and reuse the `matchesExternalLabels` to check the labels matcher.